app/models.py

```diff
@@ -172,6 +174,23 @@ def get_permissions(self, service_id=None):
             retval[service_id].append(x.permission)
         return retval
 
+    def get_organisation_permissions(self, organisation_id=None):
+        from app.dao.organisation_user_permissions_dao import organisation_user_permissions_dao
+
+        if organisation_id:
+            return [
+                x.permission
+                for x in organisation_user_permissions_dao.get_permissions_by_user_id_and_organisation_id(
+                    self.id, organisation_id
+                )
+            ]
+
+        retval = defaultdict(list)
+        for p in organisation_user_permissions_dao.get_permissions_by_user_id(self.id):
+            retval[str(p.organisation_id)].append(p.permission.value)
+
+        return retval
+
```

Comment:
Its a bit confusing that this changes the shape of what is returned based on the input parameters (either a list or a dict). But that may just be my prejudices.

Reply:
I don't really disagree, to be honest - my only thought process here was to copy what happens for `get_permissions` above (service permissions).

But in reality, `get_organisation_permissions` doesn't actually ever get called with an org_id at the moment, so I could pull this out for now?